### PR TITLE
Add to Jenkins CI

### PR DESCRIPTION
**Pull request by osharpe**
**Adding the Test stage**

Introducing the test stage into Jenkins pipeline.

**Exclude node_modules Rubocop**

I have added an exclusion for the node_modules directory as it was flagging an error.

**Karma tests for informational purposes**

I have set the Karma tests to run for information only, as there are currently no scripts attached, to prevent it failing the Jenkins job.

### CODE REVIEW

File: script/test [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+run npm install -quiet
+run bundle install --quiet
+run bundle exec bowndler install -allow-root
+run bundle exec rake db:setup
+run bundle exec rubocop .
+run bundle exec cucumber -f progress
+run bundle exec rspec -f progress
+info ./node_modules/karma/bin/karma start spec/javascripts/karma.conf.js
+info brakeman -q --no-pager --ensure-latest
```
Comment (tomas-stefano) [inline script/test]: IMHO, for fin_cap, you could add brakeman to "run" instead of adding "info". Wdyt?

Comment (osharpe) [reply script/test]: @tomas-stefano ye I agree. Most repo's have it set to "info", but considering theres no identified issues it should be "run". We just need to note tests will fail if Brakeman returns any errors/warnings. I've updated

Comment (tomas-stefano) [reply script/test]: Yes, if Brakeman fail I will be expecting the code to not be merged at all.

Comment (osharpe) [reply script/test]: @tomas-stefano Unfortunately I have had to revert back to running Brakeman test as "info". I'm having issue with locales in docker. I've come across this before and I don't have an immediate fix.

`Error: invalid byte sequence in US-ASCII While processing /var/tmp/app/app/views/shared/_nav.html.erb`
`Location: /usr/local/rvm/gems/ruby-2.4.1/gems/brakeman-4.2.0/lib/brakeman/parsers/rails3_erubis.rb:78:in `

Comment (tomas-stefano) [reply script/test]: Odd. Did you try it add the env lang as utf-8 `ENV LANG en_US.UTF-8`? There is a issue on debian (perhaps ubutu has the same problem as well) https://github.com/docker-library/ruby/issues/45

Comment (osharpe) [reply script/test]: Ye this is exactly what I was setting in the test script, which was not working. I have now set this directly in docker-compose which does the job :)

Comment (tomas-stefano) [reply script/test]: Nice!!

**Comment by pmcostello** [comment]
👍

**Pull request merged**